src/lib/reasoners/rel_utils.ml

```diff
@@ -388,16 +370,19 @@ struct
     (** Map from leaves to the *tracked* representatives that contains them *)
   }
 
+  type _ Uf.id += Id : t Uf.id
```

Comment:
I am not convinced — `kind` makes it seem like there can be multiple domain modules of the same kind but the goal is that each domain has a single unique ID distinct from that of all other domain modules.

What I think of as a "domain's kind" is rather the module itself, and the `Id` constructor is just an identifier for the module (i.e. an identifier for the kind), so it would be weird to have a kind for the kind — if that makes sense.